packages/langium-cli/src/parser-validation.ts

```diff
@@ -18,19 +14,11 @@ export function validateParser(grammar: Grammar, config: LangiumConfig, grammarC
         ...grammarConfigMap.get(grammar)?.chevrotainParserConfig,
         skipValidations: false
     };
-    const generatedSharedModule: Module<LangiumSharedServices, LangiumGeneratedSharedServices> = {
-        AstReflection: interpretAstReflection(grammar, documents),
-    };
-    const generatedModule: Module<LangiumServices, LangiumGeneratedServices> = {
-        Grammar: () => grammar,
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        LanguageMetaData: () => grammarConfigMap.get(grammar) as any,
-        parser: {
-            ParserConfig: () => parserConfig
-        }
-    };
-    const shared = inject(createDefaultSharedModule(), generatedSharedModule);
-    const services = inject(createDefaultModule({ shared }), generatedModule);
+    const services = createServicesForGrammar({
+        grammar,
+        languageMetaData: languageConfigToMetaData(grammarConfigMap.get(grammar)!),
+        parserConfig
+    });
```

Comment:
We could also pass in the grammar services here so we don't have to recreate them for every language (relevant when the CLI runs for multiple languages).